Cesium.CodeGen/Ir/Expressions/BinaryOperators/ArithmeticBinaryOperatorExpression.cs

```diff
@@ -31,9 +30,21 @@ public override void EmitTo(IDeclarationScope scope)
 
         Left.EmitTo(scope);
         EmitConversion(scope, leftType, type);
+        if (rightType is PointerType rightPointerType)
+        {
+            scope.Method.Body.Instructions.Add(Instruction.Create(OpCodes.Ldc_I4, scope.CTypeSystem.GetTypeSize(rightPointerType.Base)));
+            EmitConversion(scope, scope.CTypeSystem.Int, type);

```

Comment:
Now wait a minute. Are you converting a pointer to an `Int32` here? This is problematic!

I suggest introducing a special conversion to `nint` here.

Reply:
I'm converting expression which would be added to pointer. nit: we don't have `nint`, probably closer alternative in C would be `ptrdiff_t`. Internally it would be `nint`, but we express conversion in term of C language here.
This is probably place where I would like some changes in the expression tree. for example have `PointerAdditionOperatorExpression` which will contains all these conversions encoded (including implicit conversion as we have here), so we would not complicate simple math operations.

Reply:
I believe that we should introduce `nint` to our type system. Yes, it isn't a real primitive C type, but it implicitly exists for this exact purpose (to express types of expressions added/subtracted to/from pointers).

Reply:
I beleive this is `intptr_t`/`ptrdiff_t` which is optional typedef. Question does not `nint` or `nuint` ?  It seems that we use `nuint` based on this https://github.com/ForNeVeR/Cesium/pull/262/files#diff-e147e0aabdd129cb04f0d5838b89426531b2226cc6fcd5397a434aed085366d4R5

Reply:
I'd use `nint` for now, perhaps. Though considering how .NET handles overflows, I'm not sure this particular choice will matter much.

The choice of type size will matter, but signed vs unsigned in this context – _probably_ not.